### DISCUSSION TITLE
Update example config to 4.0

### DISCUSSION THREAD

**Discussion by Lithimlin** [rabbitmq/rabbitmq-server#12377] [Questions] [answered]
### Is your feature request related to a problem? Please describe.

A system update just broke my RabbitMQ due to a sudden update to 4.0.
I tried looking for a new example config to see how it changed and how I have to set up my AMQP 0-9-1 (or whether I can use 1.0 as a drop-in replacement), but the example is still for version 3.13.

Sorry if this seems a bit scattered, I'm scrambling to fix our deployment and haven't had time to read up on AMQP 1.0 before.

### Describe the solution you'd like

An update of the [example config file](https://github.com/rabbitmq/rabbitmq-server/blob/v3.12.x/deps/rabbit/docs/rabbitmq.conf.example)

**Comment by lukebakken** [comment]
The example configuration file is still applicable to version 4.

**Reply by Lithimlin** [reply]
See below. The anonymous login / default users config changed to the general auth_mechanism `ANONYMOUS`

**Comment by Lithimlin** [comment, marked as answer]
Okay, this was a much easier fix than I thought.
Going through the logs and upgrade notice, I found that only the guest user / anonymous login configuration changed:

```conf
## Set the default user name and password used for anonymous connections (when client
## provides no credentials). Anonymous connections are highly discouraged!
##
# mqtt.default_user = guest
# mqtt.default_pass = guest

## Enable anonymous connections. If this is set to false, clients MUST provide
## credentials in order to connect. See also the mqtt.default_user/mqtt.default_pass
## keys. Anonymous connections are highly discouraged!
##
# mqtt.allow_anonymous = true
```

This is the section that is now broken:
`default_user` and `default_pass` are now handled by the `anonymous_login_*` (see [here](https://www.rabbitmq.com/docs/next/production-checklist#anonymous-login))

**Reply by lukebakken** [reply]
@michaelklishin @ansd is this just an oversight in the example `rabbitmq.conf` to be addressed, or are there other places to be updated? For instance, here - https://github.com/rabbitmq/rabbitmq-server/releases/tag/v4.0.1

**Reply by ansd** [reply]
@lukebakken this change is clearly outlined under the Breaking Changes section in the 4.0 release notes: https://github.com/rabbitmq/rabbitmq-server/blob/main/release-notes/4.0.1.md#mqtt

> is this just an oversight in the example rabbitmq.conf to be addressed

Which `rabbitmq.conf` exactly are you referring to? Can you provide a link?

**Reply by lukebakken** [reply]
Aha! The example conf is here, which is updated: https://github.com/rabbitmq/rabbitmq-server/blob/v4.0.x/deps/rabbit/docs/rabbitmq.conf.example

@Lithimlin linked to an older version, never mind.

**Reply by lukebakken** [reply]
I read right over the MQTT section many times on the GitHub release page 🙄 

https://github.com/rabbitmq/rabbitmq-server/releases/tag/v4.0.1

![image](https://github.com/user-attachments/assets/acd9ee55-a405-44c1-88a9-c7d31ca8b1ba)

**Reply by michaelklishin** [reply]
@ansd likely the example file is [`rabbitmq.conf.example`](https://github.com/rabbitmq/rabbitmq-server/blob/main/deps/rabbit/docs/rabbitmq.conf.example), or at least I don't think we have any others :)

**Reply by ansd** [reply]
Yes, I deleted the old setting from that file.

FWIW the AMQP 1.0 settings was still documented. I just pushed a commit to `main` which deletes this one as well: https://github.com/rabbitmq/rabbitmq-server/commit/12451199728c48ddcdf19507bf5bf21cbb6d2ba7

**Reply by Lithimlin** [reply]
> @Lithimlin linked to an older version, never mind.

Ah, that's the version that's currently linked in the docs at https://www.rabbitmq.com/docs/configure#config-file
So that's probably the issue. I didn't notice the version tag in the file path

**Reply by lukebakken** [reply]
Yes, we should absolutely update that link. Thanks for pointing it out.

**Reply by ansd** [reply]
Yes, thanks! https://github.com/rabbitmq/rabbitmq-website/pull/2081
